### PR TITLE
Add back the DisplayName setting

### PR DESCRIPTION
**Pull request by Tratcher**
#391 @PinpointTownes @DamianEdwards

### CODE REVIEW

File: samples/IISSample/Startup.cs [hunk 1]
```diff
@@ -15,13 +17,14 @@ public void ConfigureServices(IServiceCollection services)
             // These two middleware are registered via an IStartupFilter in UseIISIntegration but you can configure them here.
             services.Configure<IISOptions>(options =>
             {
+                options.AuthenticationDisplayName = "Windows Auth";
```
Comment (HaoK) [inline samples/IISSample/Startup.cs]: Should just default this to the scheme like everything else to be consistent: IISDefaults.AuthenticationScheme.  We don't suffix anything else with Auth today

Comment (PinpointTownes) [inline samples/IISSample/Startup.cs]: Not sure about that. Do you want the Windows scheme to be included by default in the "providers list"?

Pretty sure this has already been discussed in the past.

Comment (Tratcher) [inline samples/IISSample/Startup.cs]: Yeah, it's null to avoid showing up by default. This is just the sample.

**Comment by PinpointTownes** [comment]
@Tratcher thanks!

Did you consider directly exposing `AuthenticationSchemeBuilder` so you could also change the scheme name?

Something like:

```csharp
public AuthenticationSchemeBuilder Scheme { get; } = new AuthenticationSchemeBuilder
{
    HandlerType = typeof(AuthenticationHandler),
    Name = IISDefaults.AuthenticationScheme
};
```

```csharp
services.Configure<IISOptions>(options =>
{
    options.Scheme.DisplayName = "Intranet";
});
```

**Comment by HaoK** [comment]
That seems more dangerous, I don't think we want to allow people to change the HandlerType at all

**Comment by Tratcher** [comment]
That's interesting, bug I agree exposing the handler would be problematic.

**Comment by HaoK** [comment]
I don't think making the scheme name configurable is that critical, but if we want to expose that as well, its harmless and matches what Security auth does

**Comment by PinpointTownes** [comment]
Well, if that's your only concern, you still can add a check to ensure the type wasn't changed before calling `Buid()`:

```csharp
if (!typeof(AuthenticationHandler).IsAssignableFrom(_options.Scheme.HandlerType))
{
    throw new InvalidOperationException("You're doing something wrong, bro'.");
}
```

**Comment by HaoK** [comment]
That just seems like a more error prone alternative to exposing only AuthenticationScheme / DisplayName

**Comment by Tratcher** [comment]
The structure would be nice..

**Comment by PinpointTownes** [comment]
Sure. But that was a clever way to save one property while adopting the same pattern as the one you've used for `CookieBuilder`... :trollface:

**Comment by HaoK** [comment]
Yeah but unlike CookieBuilder, the Handler IS the main thing, so throwing when you set that defeats the whole purpose of exposing that.

**Comment by HaoK** [comment]
I would go minimal, we can always add the builder later if things get more complicated.

**Pull request merged**